dask/bytes/core.py

```diff
@@ -194,56 +132,167 @@ def read_bytes(urlpath, delimiter=None, not_zero=False, blocksize=2**27,
     10kB sample header and list of ``dask.Delayed`` objects or list of lists of
     delayed objects if ``fn`` is a globstring.
     """
-    if compression is not None and compression not in compress_files:
-        raise ValueError("Compression type %s not supported" % compression)
+    fs, names, myopen = get_fs_paths_myopen(urlpath, compression, 'rb',
+                                            None, **kwargs)
+    if len(names) == 0:
+        raise IOError("%s resolved to no files" % urlpath)
+    sizes = [fs.size(f) for f in names]
```

Comment:
This should be the size of the file on disc. If using compression, you shouldn't be splitting anyway. The only compression that supports blocked reading is the xz code I started. To choose byte chunks out of a compressed file means scanning through the whole file to get to the appropriate part. You remember the issue on s3 files when using the get_size method that used `seek(0, 2)`? That took a long time.